mne/simulation/raw.py

```diff
@@ -0,0 +1,619 @@
+    dev_head_ts : list
+        List of device<->head transforms.
+    n_jobs : int
+        Number of jobs to run in parallel.
+    bem_blink : dict | None
+        Spherical BEM to use for blink (and ECG) simulation.
+    chpi_rrs : ndarray | None
+        CHPI dipoles to simulate (magnetic dipoles).
+    blink_rrs : ndarray | None
+        Blink dipoles to simulate.
+    ecg_rrs : ndarray | None
+        ECG dipoles to simulate.
+
+    Returns
+    -------
+    fwd : generator
+        A generator for each forward solution in dev_head_ts.
+
+    Notes
+    -----
+    Some of the forward solution calculation options from the C code
+    (e.g., `--grad`, `--fixed`) are not implemented here. For those,
+    consider using the C command line tools or the Python wrapper
+    `do_forward_solution`.
+    """
+
+    mri_head_t, mri = _get_mri_head_t(mri)
+    assert mri_head_t['from'] == FIFF.FIFFV_COORD_MRI
+
+    if not isinstance(src, string_types):
+        if not isinstance(src, SourceSpaces):
+            raise TypeError('src must be a string or SourceSpaces')
+    else:
+        if not op.isfile(src):
+            raise IOError('Source space file "%s" not found' % src)
+    if isinstance(bem, dict):
+        bem_extra = 'dict'
+    else:
+        bem_extra = bem
+        if not op.isfile(bem):
+            raise IOError('BEM file "%s" not found' % bem)
+    if not isinstance(info, (dict, string_types)):
+        raise TypeError('info should be a dict or string')
+    if isinstance(info, string_types):
+        info = read_info(info, verbose=False)
+
+    # set default forward solution coordinate frame to HEAD
+    # this could, in principle, be an option
+    coord_frame = FIFF.FIFFV_COORD_HEAD
+
+    # Report the setup
+    logger.info('Setting up forward solutions')
+
+    # Read the source locations
+    if isinstance(src, string_types):
+        src = read_source_spaces(src, verbose=False)
+    else:
+        # let's make a copy in case we modify something
+        src = src.copy()
+    nsource = sum(s['nuse'] for s in src)
+    if nsource == 0:
+        raise RuntimeError('No sources are active in these source spaces. '
+                           '"do_all" option should be used.')
+    logger.info('Read %d source spaces a total of %d active source locations'
+                % (len(src), nsource))
+
+    # Make a new dict with the relevant information
+    mri_id = dict(machid=np.zeros(2, np.int32), version=0, secs=0, usecs=0)
+    info = dict(nchan=info['nchan'], chs=info['chs'], comps=info['comps'],
+                ch_names=info['ch_names'],
+                mri_file='', mri_id=mri_id, meas_file='',
+                meas_id=None, working_dir=os.getcwd(),
+                command_line='', bads=info['bads'])
+    info = Info(info)
+
+    # Only get the EEG channels here b/c we can do MEG later
+    _, _, eegels, _, eegnames, _ = \
+        _prep_channels(info, False, True, True, verbose=False)
```

Comment:
Add checks to make sure we have EEG channels before doing the processing. Do the same for MEG.